_posts/2021-01-12-magic-control.adoc

```diff
@@ -0,0 +1,281 @@
+---
+layout: post
+title: Bored with magic tricks?
+date: 2021-01-06 # Update the date before publication
+tags: vertx
+synopsis: How to control the amount of magic from Quarkus
+author: cescoffier
+---
+
+Just before my PTO, someone told me: '_I don't like magic._'
+In this context, _magic_ refers to the amount of hidden _stuff_ done by Quarkus under the hood for the sake of simplicity.
+It includes dependency injection, annotations, and so on.
+
+It's not the first time I get that kind of comment.
+And coming from the Vert.x project, it makes sense.
+Vert.x has almost no magic, and for a good reason: too much magic can be terrible and make production tuning utterly expensive.
+Sometimes you want to have more control and avoid unexpected behavior: execute the code you wrote, and nothing else.
+
+But let's come back to Quarkus.
+You may think that Quarkus has a lot of magic tricks.
+That's not wrong, but you can decide the amount of magic you want and the amount of control you feel comfortable with.
+You don't have to use dependency injection or managed clients if you prefer doing things yourself.
+
+In this post, we will cover three different approaches to reducing the amount of magic.
+We will go from almost no magic to just enough to get a good developer experience.
+Examples from this blog post are available on https://github.com/cescoffier/quarkus-magic-control[GitHub].
+
+== The _almost no magic_ approach
+
+Quarkus applications are Java applications.
+So, somewhere there is a `public static void main(String... args)`.
+While in general, when you are using Quarkus, you don't need to write that method; it can be convenient and give you more control about your application's startup.
```

Comment:
```suggestion
While you don't need to write that method when using Quarkus, it can still be convenient and give you more control about your application startup.
```